File: backend/segmentation_fault/schemas/posts.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class Post(BaseModel):
    id: int
    created: datetime
    updated: Optional[datetime]
    author: int
    title: str
    content: str


class PostCreate(BaseModel):
    title: str
    content: str


class PostUpdate(BaseModel):
    title: str
    content: str

File: frontend/segmentation_fault/api/schemas/posts.ts
export interface Post {
    id: number;
    created: Date;
    updated?: Date;
    author: number;
    title: string;
    content: string;
    votes: number;
    vote: boolean | null;
}

export interface PostCreate {
    title: string;
    content: string;
}

export interface PostCreateExternal {
    title: string;
    content: string;
}

export interface PostExternal {
    id: number;
    created: string;
    updated?: string;
    author: number;
    title: string;
    content: string;
}

export interface PostUpdate {
    title: string;
    content: string;
}

export interface PostUpdateExternal {
    title: string;
    content: string;
}
